internal/services/aadgraph/user_resource.go

```diff
@@ -118,27 +118,27 @@ func resourceUserCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	userCreateParameters := graphrbac.UserCreateParameters{
-		AccountEnabled: p.BoolI(d.Get("account_enabled")),
-		DisplayName:    p.StringI(d.Get("display_name")),
+		AccountEnabled: utils.BoolI(d.Get("account_enabled")),
+		DisplayName:    utils.StringI(d.Get("display_name")),
 		MailNickname:   &mailNickName,
 		PasswordProfile: &graphrbac.PasswordProfile{
-			ForceChangePasswordNextLogin: p.BoolI(d.Get("force_password_change")),
-			Password:                     p.StringI(d.Get("password")),
+			ForceChangePasswordNextLogin: utils.BoolI(d.Get("force_password_change")),
+			Password:                     utils.StringI(d.Get("password")),
 		},
 		UserPrincipalName: &upn,
 	}
 
 	if v, ok := d.GetOk("usage_location"); ok {
-		userCreateParameters.UsageLocation = p.StringI(v)
+		userCreateParameters.UsageLocation = utils.StringI(v)
 	}
 
 	if v, ok := d.GetOk("immutable_id"); ok {
-		userCreateParameters.ImmutableID = p.StringI(v)
+		userCreateParameters.ImmutableID = utils.StringI(v)
```

Comment:
(As before) we should use types here rather than passing around interface which is a code smell / anti pattern